com.zsmartsystems.zigbee/src/main/java/com/zsmartsystems/zigbee/app/discovery/ZigBeeNodeServiceDiscoverer.java

```diff
@@ -203,16 +203,12 @@ private void startDiscovery(Set<NodeDiscoveryTask> newTasks) {
         synchronized (discoveryTasks) {
             logger.debug("{}: Node SVC Discovery: starting new tasks {}", node.getIeeeAddress(), newTasks);
 
-            // Remove any tasks that we know are not supported by this device
-            if ((!supportsManagementLqi || node.getNodeDescriptor() != null
-                    && node.getNodeDescriptor().getLogicalType() == LogicalType.UNKNOWN)
-                    && newTasks.contains(NodeDiscoveryTask.NEIGHBORS)) {
+            // Remove router/coordinator-only tasks if the device is possibly an end node.
+            final boolean isPossibleEndDevice = isPossibleEndDevice();
+            if ((!supportsManagementLqi || isPossibleEndDevice)) {
```

Comment:
By changing the logic around here it should not be needed to add the additional ```if supported``` checks in the tasks themselves.

Reply:
Stet.

Since a possible end device may  actually not be an end device, whether or not the requests are supported may not be known a priori.  I think support is mandatory for ZR and ZC, but it's in the stack profile, so I'm not sure.